lua/entities/gmod_wire_expression2/core/ranger.lua

```diff
@@ -77,8 +77,8 @@ local function ranger(self, rangertype, range, p1, p2, hulltype, mins, maxs, tra
 		tracedata.start = Vector( p1[1], p1[2], p1[3] )
 		tracedata.endpos = Vector( p2[1], p2[2], p2[3] )
 	elseif rangertype == 3 then
-		tracedata.start = Vector( p1[1], p1[2], p1[3] )
-		tracedata.endpos = tracedata.start + Vector( p2[1], p2[2], p2[3] ):GetNormalized()*range
+		tracedata.start = WireLib.clampPos( Vector( p1[1], p1[2], p1[3] ) )
+		tracedata.endpos = WireLib.clampPos( tracedata.start + Vector( p2[1], p2[2], p2[3] ):GetNormalized() * range )
```

Comment:
My previous understanding was that `Vector(huge, huge, huge):GetNormalized()` could crash here. @brandonsturgeon was that the case?

Reply:
No, it was the trace itself causing the deadlock. I believe that will produce nan. Idk if nan will be clamped correctly or deadlock too tho.

Reply:
Nan doesn't seem to cause any issue